backend/app/infrastructure/routes/rewards.py

```diff
@@ -217,10 +247,42 @@ def get(self, epoch: int):
 
 @ns.route("/estimated_budget")
 @ns.doc(
-    description="Returns estimated rewards budget available when GLM locked by given period of time"
+    description="Returns estimated rewards budget available when GLM locked by given number of full epochs"
 )
 class EstimatedUserBudget(OctantResource):
     @ns.expect(estimated_budget_request)
+    @ns.marshal_with(user_budget_with_matched_funding_model)
+    @ns.response(200, "Budget successfully retrieved")
+    def post(self):
+        no_epochs, glm_amount = ns.payload["numberOfEpochs"], int(
+            ns.payload["glmAmount"]
+        )
+        app.logger.debug(
+            f"Getting user estimated budget for {no_epochs} epochs and {glm_amount} GLM. Getting matched_funding based on previous epoch."
+        )
+
+        current_epoch_num = epochs.get_current_epoch()
+        validate_estimate_budget_by_epochs_inputs(no_epochs, glm_amount)
+        validate_if_given_epoch_has_previous_one(current_epoch_num)
+
+        rewards = estimate_rewards(current_epoch_num - 1, no_epochs, glm_amount)
+        budget = rewards.estimated_budget
+        leverage = rewards.leverage
+        matched_funding = rewards.matching_fund * leverage
+
+        app.logger.debug(
+            f"Estimated user budget: {budget}, estimated matched_funding: {matched_funding} with leverage: {leverage}"
+        )
+
+        return {"budget": budget, "matchedFunding": matched_funding}
+
+
+@ns.route("/estimated_budget/by_days")
```

Comment:
We don't use this endpoint anymore for now but I've left it here as not to remove controller logic that's based on any value for lockDuration for current epoch and it may be useful in the future.